araproc/analysis/rpr.py

```diff
@@ -0,0 +1,206 @@
+import numpy as np
+from scipy.ndimage import uniform_filter1d
+from araproc.framework import waveform_utilities as wfu
+
+
+def get_max_info(channel_wf, channel_time, use_debug=False):
+    """
+    Identifies the maximum voltage and its corresponding time.
+
+    Parameters
+    ----------
+    channel_wf: numpy array
+        Array of voltage values (squared waveform data).
+    channel_time: numpy array
+        Array of time values corresponding to the waveform data.
+    use_debug: bool, optional
+        If True, prints debug information.
+
+    Returns
+    -------
+    max_bin: int
+        Index of the maximum voltage in the waveform.
+    max_val: float
+        Maximum voltage value in the waveform.
+    max_time: float
+        Time corresponding to the maximum voltage value.
+    """
+    max_bin = np.nanargmax(channel_wf)
+    max_val = channel_wf[max_bin]
+    max_time = channel_time[max_bin]
+
+    if use_debug:
+        print(f"Max value: {max_val}, Max time: {max_time}")
+
+    return max_bin, max_val, max_time
+
+def get_sqrt_volt_sum_wf(channel_wf, dt):
+    """
+    Smooth the squared voltage waveform using a rolling mean filter.
+
+    Parameters
+    ----------
+    channel_wf: numpy array
+        Array of squared voltage values.
+    dt: float
+        Step size from time array.
+
+    Returns
+    -------
+    smoothed_waveform: numpy array
+        Smoothed version of the input waveform.
+    """
+    sum_win = 25
+    sum_win_idx = int(np.round(sum_win / dt))
+
+    return np.sqrt(uniform_filter1d(channel_wf, size=sum_win_idx, mode='constant'))
+
+def get_mean_sigma_in_no_max(channel_wf, max_bin, wf_len, use_debug=False):
+    """
+    Computes the mean and standard deviation of the waveform, excluding the region
+    around the maximum voltage.
+
+    Parameters
+    ----------
+    channel_wf: numpy array
+        Smoothed voltage values of the waveform.
+    max_bin: int
+        Index of the maximum voltage in the waveform.
+    wf_len: int
+        Total number of samples in the waveform.
+    use_debug: bool, optional
+        If True, prints debug information.
+
+    Returns
+    -------
+    noise_mean: float
+        Mean voltage excluding the peak window.
+    noise_sigma: float
+        Standard deviation excluding the peak window.
+    """
+    bin_8 = wf_len // 8
+    pad_no_max = channel_wf.copy()
+    front_idx = max(0, int(max_bin - bin_8))
+
+    pad_no_max[front_idx:max_bin + bin_8 + 1] = np.nan
+    noise_mean = np.nanmean(pad_no_max)
+    noise_sigma = np.nanstd(pad_no_max)
+
+    if use_debug:
+        print(f"Mean: {noise_mean}, Sigma: {noise_sigma}")
+
+    return noise_mean, noise_sigma
+
+def get_ch_sliding_v2_snr_uw(channel_wf, channel_time, wf_len, use_debug=False):
```

Comment:
Sorry this is a little nitpicky, but can we simplify this function name? Actually given that it returns the rpr itself, maybe this should all be put directly into `run_rpr_calculation`?